testutils/src/main/java/org/zaproxy/zap/testutils/ScanRuleTests.java

```diff
@@ -128,28 +130,59 @@ default void shouldHaveValidReferences() {
 
             if (!HttpHeader.HTTPS.equals(uri.getScheme())) {
                 errors.add(AlertReferenceError.Cause.NOT_HTTPS.create(reference, ""));
-            } else if (false) {
-                fetchUrl(uri, reference, errors);
+            } else {
+                assumingThat(
+                        "1".equals(System.getenv("ZAP_REMOTE_TESTS")),
+                        () -> fetchUrl(uri, reference, errors));
             }
         }
 
         assertThat(errors.toString(), errors, is(empty()));
```

Comment:
Should we call `addErrorIfNotAllowed` in this method too? (e.g. if a test wanted to ignore `NOT_LINK` for any reason)

Reply:
Can do, but do we need that? We have the "rule" of references being HTTPS links, they are not supposed to be free text (I guess they could be HTTP but we don't have any).